### PR TITLE
Shuna51

### PR DESCRIPTION
**Pull request by JoannaCode**
@FreeTymeKiyan Finished two questions about NQueen

### CODE REVIEW

File: Hard/NQueens_shuna.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+public class NQueens {
+    public static List<List<String>> solveNQueens(int n) {
+        List<List<String>> res = new ArrayList<List<String>>();
+        int[] board = new int[n];
+        QueenDFS(n, 0, board, res);
+        return res;
+    }
+
+    public static void QueenDFS(int nrow, int row, int[] board, List<List<String>> res){
```
Comment (FreeTymeKiyan) [inline Hard/NQueens_shuna.java]: QueenDFS -> queenDfs, please follow the camel case naming convention. 
nrow is the last row or nth row, right? Might use another clearer variable name.

File: Hard/NQueens_shuna.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        if(nrow == row){
+            List<String> s1 = new ArrayList<String> ();
```
Comment (FreeTymeKiyan) [inline Hard/NQueens_shuna.java]: guess you forgot to format your code again.

File: Hard/NQueens_shuna.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * N-Queens II
+     * Follow up for N-Queens problem.
+     * Now, instead outputting board configurations, return the total number of distinct solutions.
+     */
+
+    public static int totalNQueens(int n) {
+        int res[] = {0};
```
Comment (FreeTymeKiyan) [inline Hard/NQueens_shuna.java]: really don't understand why you need an integer array here, could you explain?

Comment (JoannaCode) [inline Hard/NQueens_shuna.java]: We can make this res as global variable. But in fact, we do not want the global variable in practice (em..in my company 's code rule). And if we make it local ,it won't be saved in recursive(I tried). So I choose the array.

**Comment by JoannaCode** [comment]
@FreeTymeKiyan Em...I think the function name should be queenDFS. DFS is Depth-First-Search.

**Comment by FreeTymeKiyan** [comment]
Yeah, I know DFS is the abbreviation of Depth-First Search. Still, we should use Dfs in function names, as that is specially pointed out in camel case convention, although it's not widely adopted and noticed.

**Comment by FreeTymeKiyan** [comment]
:+1:

**Pull request merged**